### PR TITLE
Compatibility pass on flutter/rendering tests for JavaScript compilation. (5)

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Flags certain test cases in rendering using package:test tags in order to facilitate landing support for running all framework tests in a browser.

Requires https://github.com/flutter/flutter/pull/33349

### CODE REVIEW

File: packages/flutter/test/rendering/debug_test.dart [hunk 1]
```diff
@@ -15,10 +15,10 @@ void main() {
     final Matrix4 identity = Matrix4.identity();
     final List<String> description = debugDescribeTransform(identity);
     expect(description, equals(<String>[
-      '[0] 1.0,0.0,0.0,0.0',
-      '[1] 0.0,1.0,0.0,0.0',
-      '[2] 0.0,0.0,1.0,0.0',
-      '[3] 0.0,0.0,0.0,1.0',
+      ignoreWebNumericQuirks('[0] 1.0,0.0,0.0,0.0'),
```
Comment (Hixie) [inline packages/flutter/test/rendering/debug_test.dart]: What is ignoreWebNumericQuirks?

Comment (jonahwilliams) [reply packages/flutter/test/rendering/debug_test.dart]: Added the implementation here: https://github.com/flutter/flutter/pull/33408

Comment (yjbanov) [reply packages/flutter/test/rendering/debug_test.dart]: Alternatively, we could implement `debugDescribeTransform` such that it produces identical formatting. This is debug code anyway, so it won't affect performance.

Comment (Hixie) [reply packages/flutter/test/rendering/debug_test.dart]: Seems like we could solve this by finally fixing https://github.com/flutter/flutter/issues/1221

Comment (jonahwilliams) [reply packages/flutter/test/rendering/debug_test.dart]: I spoke with @ferhatb today and they already have code that needs to be upstreamed to fix this

Comment (jonahwilliams) [reply packages/flutter/test/rendering/debug_test.dart]: Err, this particular case and not the more general isssue :)

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 2]
```diff
@@ -479,6 +479,8 @@ Future<void> main() async {
   // we don't have an embedder yet this is the lowest-most layer we can put
   // this stuff in.
   await ui.webOnlyInitializeEngine();
+  (ui.window as dynamic).debugOverrideDevicePixelRatio(3.0);
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: This correctly sizes the window to match the flutter tester default

**Comment by jonahwilliams** [comment]
Updated to address comments and to run on CI

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart (hunk 2 above)
Comment (yjbanov) [reply packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: We should probably leave a `TODO` to remove the need for `dynamic`.

/cc @hterkelsen

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**